### ISSUE TITLE
Graph Data Science "Node Similarity" algorithm documentation is partially unclear

### ISSUE THREAD

**Issue by aaramg** [neo4j/graph-data-science#96]
In https://neo4j.com/docs/graph-data-science/current/algorithms/node-similarity/ similarity algorithm documentation there are a lot of config properties in the "General configuration for algorithm execution on an anonymous graph" section, that are not well explained.
For example:

"nodeProjection", "relationshipProjection", "nodeQuery", "relationshipQuery", "nodeProperties","relationshipProperties" etc. 
How to use them? 
What should those query properties (nodeQuery, relationshipQuery) return?? 
How they work?? 
Are they for filtering?
Could not find examples explaining their usage (besides "nodeProjection", "relationshipProjection" properties, which are used in the dedicated graph creation configs example in the same doc page, but would be great to see more capabilities of those properties too).
"relationshipWeightProperty" configuration property is used in the example in the end, but if we search "relationshipWeightProperty" word on the same page, there is nowhere written about it. It is being clear just logically from the example. Maybe there are other properties that are also acceptable but not written in the documentation???
Also, there is no chapter about the "Similarity algorithms" in the free book of "Graph Algorithms". I mean no one can find info even from that resource.

Let's take the example nodes and relationships are these (as in the example in doc page)
Also, let's also consider the "strength" field of relationship. 
Could you give an example using all those config properties in the documentation??
```
CREATE
  (alice:Person {name: 'Alice'}),
  (bob:Person {name: 'Bob'}),
  (carol:Person {name: 'Carol'}),
  (dave:Person {name: 'Dave'}),
  (eve:Person {name: 'Eve'}),

  (guitar:Instrument {name: 'Guitar'}),
  (synth:Instrument {name: 'Synthesizer'}),
  (bongos:Instrument {name: 'Bongos'}),
  (trumpet:Instrument {name: 'Trumpet'}),

  (alice)-[:LIKES]->(guitar),
  (alice)-[:LIKES]->(synth),
  (alice)-[:LIKES {strength: 0.5}]->(bongos),
  (bob)-[:LIKES]->(guitar),
  (bob)-[:LIKES]->(synth),
  (carol)-[:LIKES]->(bongos),
  (dave)-[:LIKES]->(guitar),
  (dave)-[:LIKES]->(synth),
  (dave)-[:LIKES]->(bongos);
```

**Comment by Mats-SX** [comment]
Hello @aaramg and thank you for reaching out to us.

Let me try to explain the various concepts, since you ask many questions here. First, you must understand that in GDS there are three main steps: graph projection, algorithm execution, result processing. For the sake of this issue, we can ignore the third step and focus on the first two.

**Graph projection**
All* algorithms in GDS takes a graph as primary input. The GDS graph is a projection over the Neo4j (database) graph. There are two types of these projections: [native projections](https://neo4j.com/docs/graph-data-science/current/management-ops/native-projection/) and [Cypher projections](https://neo4j.com/docs/graph-data-science/current/management-ops/cypher-projection/). These are powered by their own dedicated procedures but they have the same output: a named graph in the GDS graph catalog.

**Algorithm execution**
The projected graph from the catalog is passed in as a primary argument to the algorithm procedure, by using its name. Following that is the algorithm-specific configuration. This is the recommended syntax to work with GDS.

As an alternative, one can avoid the graph catalog and inline the graph projection step into the algorithm execution. We refer to this as using an *anonymous* projection, because there is no name for the graph. Still, these anonymous projections may be of either type: native projection or Cypher projection. However, since they are inlined, there are no dedicated procedures. Instead, you will configure the projection together with the algorithm in the configuration map.

This is where the configuration keys you are asking for comes into play. `nodeProjection`, `relationshipProjection`, `nodeProperties` and `relationshipProperties` keys are used to configure a native projection, and these have the same meaning as is explained in the native projection documentation. Similarly, `nodeQuery` and `relationshipQuery` keys are used to configure a Cypher projection, with the same meaning as is explained in the Cypher projection documentation. Note that you can not use _both_ a native and a Cypher projection, but you must choose one of these two pairs of keys.

The `relationshipWeightProperty` key is different. And for this one you have found a bug in our documentation. It should be mentioned in the algorithm-specific configuration table, but it is missing. I will make a note to add it there. I will also double-check that there aren't any other unmentioned but supported configuration keys. Thanks for finding this!

Both syntax variants are explained in some detail on the [Syntax overview page](https://neo4j.com/docs/graph-data-science/current/algorithms/syntax/).

----

So, in summary, these keys are all used to configure an anonymous projection in the anonymous syntax variant. You may ignore this syntax variant and use only the named graph variant, which is in fact what we recommend. It is not about filtering, and it does not have any relationship to the Node Similarity algorithm (or any other specific algorithm).

*with the exception of some `alpha` tier procedures.

**Comment by aaramg** [comment]
Thank you for your explanation.
But I don't understand one thing:
Let's take we have this graph:
```
CREATE
  (alice:Person {name: 'Alice'}),
  (bob:Person {name: 'Bob'}),
  (carol:Person {name: 'Carol'}),
  (dave:Person {name: 'Dave'}),
  (eve:Person {name: 'Eve'}),

  (guitar:Instrument {name: 'Guitar'}),
  (synth:Instrument {name: 'Synthesizer'}),
  (bongos:Instrument {name: 'Bongos'}),
  (trumpet:Instrument {name: 'Trumpet'}),

  (alice)-[:LIKES]->(guitar),
  (alice)-[:LIKES]->(synth),
  (alice)-[:LIKES {strength: 0.5}]->(bongos),
  (bob)-[:LIKES]->(guitar),
  (bob)-[:LIKES]->(synth),
  (carol)-[:LIKES]->(bongos),
  (dave)-[:LIKES]->(guitar),
  (dave)-[:LIKES]->(synth),
  (dave)-[:LIKES]->(bongos);
```
How can I change the query below to include only users with the name: "Alice", "Bob" and their "LIKES" only.
```
CALL gds.graph.create(
    'myGraph',
    ['Person', 'Instrument'],
    {
        LIKES: {
            type: 'LIKES',
            properties: {
                strength: {
                    property: 'strength',
                    defaultValue: 1.0
                }
            }
        }
    }
);
```
The concept I don't understand is how can I narrow the graph projection to some specific little graph but filtering with **properties** of Node and Relationship. I could not find that in the documentations.
Thanks in advance.

**Comment by Mats-SX** [comment]
Filtering on property values is not (yet) possible using native projections. It is on our roadmap, but not yet selected for a target version. It is possible to do it in other ways, however.

First, a detail: you can not include the `LIKES` relationships without also including their target nodes, which are of label `Instrument` in the above graph. Assuming you meant a graph of `Alice`, `Bob` and also all `Instrument` nodes, a Cypher projection could be used like this:

```
CALL gds.graph.create.cypher(
    'myGraph',
    'MATCH (n:Person) WHERE n.name = 'Alice' OR n.name = 'Bob' RETURN id(n) AS id, labels(n) AS labels
     UNION
     MATCH (i:Instrument) RETURN id(i) AS id, labels(i) AS labels',
    'MATCH (s:Person)-[r:LIKES]->(t:Instrument) WHERE s.name = 'Alice' OR s.name = 'Bob' RETURN id(s) AS source, id(t) AS target, type(r) AS type, r.strength AS strength'
);
```

Note the `UNION` node query. You could also use a more complex `OR` predicate like `WHERE (n:Person AND (n.name = 'Alice' OR n.name = 'Bob')) OR n:Instrument`; I don't know which would perform better.

This will not be very performant for huge graphs. As an alternative, it is possible to pre-process with Cypher and persist the subgraph using a new label, and then use a native projection on the new label. In this case you will only pay the extra performance price once. To do this, run a Cypher update query such as:

```
MATCH (n) WHERE (n:Person AND (n.name = 'Alice' OR n.name = 'Bob')) OR n:Instrument
SET n:AliceBobSubgraph
```

and then use a native projection:

```
CALL gds.graph.create(
    'myGraph',
    'AliceBobSubgraph',
    {
        LIKES: {
            properties: {
                strength: {
                    defaultValue: 1.0
                }
            }
        }
    }
);
```

Hope this helps!

**Comment by s1ck** [comment]
We updated the node similarity documentation. It will be visible once we release the docs.
